### PR TITLE
Pin local chart dependencies in Chart.yaml files

### PR DESCRIPTION
**Pull request by keithhand**
## What does this PR change?

Pins local chart dependencies inside helm `Chart.yaml` files

## Does this PR rely on any other PRs?

N/A

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Allows `helm lint` command to pass.

## Links to Issues or ZD tickets this PR addresses or fixes

- #1025 

## How was this PR tested?

After successfully upgrading the helm chart I was able to also successfully run `helm lint`.

```
$ helm upgrade -f values.yaml kubecost ./cost-analyzer/
W0105 12:44:47.649368   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0105 12:44:47.654934   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0105 12:44:47.663873   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0105 12:44:47.671629   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0105 12:44:47.677892   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0105 12:44:47.685105   28904 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
Release "kubecost" has been upgraded. Happy Helming!
NAME: kubecost
LAST DEPLOYED: Wed Jan  5 12:44:46 2022
NAMESPACE: kubecost
STATUS: deployed
REVISION: 4
TEST SUITE: None
NOTES:
--------------------------------------------------Kubecost has been successfully installed. When pods are Ready, you can enable port-forwarding with the following command:
    
    kubectl port-forward --namespace kubecost deployment/kubecost-cost-analyzer 9090
    
Next, navigate to http://localhost:9090 in a web browser.

Having installation issues? View our Troubleshooting Guide at http://docs.kubecost.com/troubleshoot-install

$ helm lint ./cost-analyzer/
==> Linting ./cost-analyzer/
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

## Have you made an update to documentation?

N/A

### CODE REVIEW

**Comment by AjayTripathy** [comment]
Hey @keithhand thanks for this PR. As I recall, the "requirements" field in the Chart.yaml may break some third-party integrations including the rancher partner-charts. It's getting time for an upgrade there anyway, maybe @elenalape can hand that off to the support engineering team, then we can
1. do an upgrade in rancher's partner-charts to 1.89.1
2. merge this linting PR between 1.89.X and 1.90
3. try an upgrade again of rancher's partner charts to 1.90 and make sure this doesn't break it.

**Comment by AjayTripathy** [comment]
@keithhand following up here now that you've had a chance to look at the rancher process. Does this pr interfere with that?

**Comment by keithhand** [comment]
@AjayTripathy No, this should go through fine. I checked a few projects from the partner-charts repo, and this formatting seems acceptable since a few other projects are using it. I didn't add condition fields for each dependency, which doesn't seem necessary for the merge, but it isn't a bad idea. I can push that up after making sure things configure correctly.

File: cost-analyzer/charts/prometheus/Chart.yaml [hunk 1]
```diff
@@ -18,3 +18,7 @@ sources:
 - https://github.com/kubernetes/kube-state-metrics
 tillerVersion: '>=2.8.0'
 version: 11.0.2
+dependencies:
+  - condition: kube-state-metrics.disabled
+    name: kube-state-metrics
+    repository: file://./charts/kube-state-metrics
```
Comment (michaelmdresser) [inline cost-analyzer/charts/prometheus/Chart.yaml:24]: @AjayTripathy After our metrics emission changes, do we still need to have KSM in the Helm chart?

**Review by michaelmdresser — APPROVED** [review]
Other than my question to Ajay, this LGTM.

File: cost-analyzer/charts/prometheus/Chart.yaml (hunk 1 above)
Comment (AjayTripathy) [reply cost-analyzer/charts/prometheus/Chart.yaml:24]: Probably not but that's a pretty big change outside the scope of this PR.

File: cost-analyzer/Chart.yaml [hunk 2]
```diff
@@ -9,3 +9,13 @@ annotations:
   "artifacthub.io/links": |
     - name: Homepage
       url: https://www.kubecost.com
+dependencies:
+  - condition: global.grafana.enabled
+    name: grafana
+    repository: file://./charts/grafana
```
Comment (michaelmdresser) [inline cost-analyzer/Chart.yaml:15]: cc @nealormsbee for your (eventual) Grafana deprecation/removal

**Review by michaelmdresser — APPROVED** [review]
Still LGTM!

**Pull request merged**